### DISCUSSION TITLE
[Evolution] Component Select

### DISCUSSION THREAD

**Discussion by LucasAlbuquerque0** [Telefonica/mistica-design#1989] [Proposals]
### Goal

As the Select Desktop component, the **Select Mobile** component receives **"Status: Opened"**.

### Context

Since the mobile component does not contain the exposed Menu, some designers end up having to create their own Menu to use within visual applications, leaving room for errors and problems when searching for these components and mainly causing visual divergence during QA.

Even having the Menu component, it ends up being an extra step of knowledge that some designers end up forgetting or not knowing and tend to make mistakes.

### Impact

Problems when doing QA of the screens, where there is a discrepancy between what is presented in Figma and what is seen in the app

### Discussion Questions

To have consistency between mobile and desktop components, they need to be reflective. Do you consider the evolution valid?

### Additional considerations

Visual presentation of the problem
![image](https://github.com/user-attachments/assets/7a40bbb1-aaf4-46a7-b3fe-da694e07bcb9)

To maintain the same consistency between components, adding the same status on mobile

**Comment by yceballost** [comment]
Hi @LucasAlbuquerque0 

This is because in mobile the select opened is used with native pickers. This picker is depends on browsers and OS

Like this: 
![image](https://github.com/user-attachments/assets/327d2d1e-b276-41bf-9f93-59ab3590a80e)

**Comment by LucasAlbuquerque0** [comment]
I have two questions @yceballost 

1. An interesting suggestion might be to use something similar to Sheet's Proto Guide, which serves to show how to use the component in the best way.
![image](https://github.com/user-attachments/assets/73a3080d-7a51-4914-9eed-943a11a44334)
Mainly to avoid situations where, due to lack of knowledge, this menu is manually created.

2. Is it possible to use Sub Value in Menu Select?
3. ![image](https://github.com/user-attachments/assets/5dc94107-9e82-44f9-a631-dd24d39fb00b)

**Reply by yceballost** [reply]
Si, una opción podría ser añadir el "opened" en mobile y mostrar una "proto guide" o incluso añadir varias opciones
- Opened (iOS)
- Opened (Android)
- Opened (Web)

**Opened (iOS)**
Una "guide" como la que comentas que te explique que aparece el picker como en esta captura
![image](https://github.com/user-attachments/assets/327d2d1e-b276-41bf-9f93-59ab3590a80e)
**Opened (Web)**
![image](https://github.com/user-attachments/assets/0677cfc8-9c81-4d02-aae7-e83c81407741)
**Opened (Android)**
(no tengo captura xD)

**Reply by yceballost** [reply]
A tu segunda pregunta, invoco a @aweell porque si no me equivoco, el autocomplete podría utilizarse para eso?

Me suena haber visto un ejemplo de Alex pero no logro encontrarlo

**Reply by aweell** [reply]
Si, el select no se podría utilizar, pero puede construirse algo parecido con un text field y un menú que permite configurar como quieras cada una de las opciones.

En su momento se creó este ejemplo:

[Ejemplo](https://mistica-ntceapvhe-aweell.vercel.app/playroom#?code=N4Igxg9gJgpiBcIA8AlGBnADhAduglgG4wAyAhgJ4QCuALgHwA6OABC0gEIQAeLmZUKPhwBzAJoBeYABYAHAF8mrNuwCyMHNWYqVAJw2xdAFTK6RMWlIAUwFvoBmAGha4ACvvTpn%2BdOs0B5TA0WeQBKFgl6FittHRUkIxhuWgAxfBgAGyhYuJV7agyMgHV8KFoACxzcuxh7KQd5Kty3D3QpFox0RuVqlkIyDOoYKXNaAGVaMloYGJB0CwA1AaH0RhBQ7t62HDIAW2G1nf21priMsgAjTIk1jjIwAGsRXRocKBZIDIhdE57qsjoEAAwhBdpgMhYDiAIPZ7L9egB6JS5DanfRvGC6PzUay2BzOMDnTwAOT2MAJX3mIXCkWipzYSCEhBqdWADQ%2BRPQpP2UkJZBJZMU9PiXF4-EEwhEUgAjAA2IV-XpICb3B4sLD3YbABTIrY6YAAbWFvWAxq202S8BYt1Vz1e7wuujIbzWjjNvVg6DAunwmFo%2BFw8DW7nwu1MFA%2BEC%2BPxAbsVepU-UGMCtNseduobxYjud2Vj7uqn2%2BVqLunQADoc284wm4vIa7WVKb47WLbRUyA7umXpmHU6s-NIG9w66Cx6MN7ff7A2sxjAh1Bw5Ho6OWwmk0MO12nj2s1X3oPcIvdBRV424qWS1HvhWLrbd1AOP2oHOF%2BGG%2Bf62Om9%2BdG2t-e9osAM0y6Ds-rEGe57jl6Pp%2BgGOBBiAxIwHQToZMu3xQdBfTLCm1qdoBvbARkoHgUQMDYdBl6YWWlZEW8ACCpGYuRxAfo2X5rrkAC65ZhpgVhWBA8G4DSUQxNx1ScDwfACEIoiSNq8gsA8MAUO0ok4OWG4wAqOEMkC5Tzg8Fw8L%2BcRHFqInTtpbabAZbC4EZzrmNY4ksM2jk6PM4yTNMsy%2BUsyarLGLhaTpeGhBSEDzFYoQANwWTo8gOY5YDGY8MBQCMFgqgFaxBXhoXOAaPGhOWwiEtQnrCRFumolJerooYuKRjgtAvBkACiEL7B1zjnFcGQAJLvGEERRF53lsPotDUGBdJNY2yqTI86r8GAWrSPpM1xEgI04BkwgwBtmpSAoJH4CIOAjdMuxtGsW0dZiay6nt%2B1AvguiEpRy2OXe3b2iC0aabZ5almlH1xAQABeWpylD0MsPQSAIl9P0Qu9yPsIkyQAMwsKUUhDZkY2KMANkIeW9lo3jtD49jH1o4dx04DATN7QkSS0AATETOXAKTo1QCppa8tedFtq%2BR7hip9CUxFnqTlp8h0zzvOc45aMqo8WvQYlyUqKlRtIhZaOivrKKNbWOtrQ8VvsAilvCmjTKczbKhm8oFs8EoaNoFguAEMQ5BUHQSixiAFQwPs6AIAaICqBAhA%2BJMuggDxjggAA7qUFTx-ABrSgADLz0g8fIQA)

No se si es al que te refieres?

**Reply by LucasAlbuquerque0** [reply]
@yceballost En cuanto a Select Opened. ¿cómo procedemos con la idea?
¿Genero una alternativa o simplemente creo un issue en git solicitando la evolución de la idea?

**Reply by LucasAlbuquerque0** [reply]
@aweell  Eso es exactamente, era más saber si era posible. Esperaremos a que se defina o no la necesidad. Muchas gracias Alex 😉

**Reply by yceballost** [reply]
Si por favor, crea una issue en Jira de OBVIVO, aunque el componente existe en código lo tenemos dentro de Mística Lab y lo ideal sería "oficializarlo" con specs y demás.

https://mistica-web.vercel.app/?path=/story/m%C3%ADstica-lab-autocomplete--default

**Reply by LucasAlbuquerque0** [reply]
@yceballost Qué tal la idea de tener un Select Opened con una guía proto para su uso ¿Creo un Issue de la idea?

**Reply by yceballost** [reply]
Sí genial! crea una issue para eso :)

**Reply by LucasAlbuquerque0** [reply]
Mostrar, crear la solicitud:
https://jira.tid.es/browse/OBVIVO-2909

**Comment by LucasAlbuquerque0** [comment]
Hola, Yayo (@yceballost) 
¿Todo está bien?

Solo para reforzar que creé esta demanda en Jira España
https://jira.tid.es/browse/OBVIVO-2909

**Reply by yceballost** [reply]
Perfecto! priorizarlo con Julio para Q2 :)

**Reply by yceballost** [reply]
Podemos cerrar esta discussión, verdad?

**Reply by LucasAlbuquerque0** [reply]
Claro que podemos cerrar

**Reply by yceballost** [reply]
La he dividido en 2:

- https://github.com/Telefonica/mistica-design/issues/2089
- https://github.com/Telefonica/mistica-design/issues/2088

La 2088 si quieres es algo que podrías incluir tu mismo cuando quieras, te abrimos una rama en la librería y lo añades :) (si no, lo metemos para Q2 como hemos dicho)
